lib/src/androidTest/kotlin/at/bitfire/ical4android/util/AndroidTimeUtilsTest.kt

```diff
@@ -277,45 +277,98 @@ class AndroidTimeUtilsTest {
     @Test
     fun testAndroidStringToRecurrenceSets_UtcTimes() {
         // list of UTC times
-        val exDate = AndroidTimeUtils.androidStringToRecurrenceSet("20150101T103010Z,20150702T103020Z", false) { ExDate(it) }
-        assertNull(exDate.timeZone)
-        val exDates = exDate.dates
-        assertEquals(Value.DATE_TIME, exDates.type)
-        assertTrue(exDates.isUtc)
-        assertEquals(2, exDates.size)
-        assertEquals(1420108210000L, exDates[0].time)
-        assertEquals(1435833020000L, exDates[1].time)
+        AndroidTimeUtils.androidStringToRecurrenceSet(
+            "20150101T103010Z,20150702T103020Z",
+            false
+        ) { ExDate(it) }.forEach { exDate ->
+            assertNull(exDate.timeZone)
+            val exDates = exDate.dates
+            assertEquals(Value.DATE_TIME, exDates.type)
+            assertTrue(exDates.isUtc)
+            assertEquals(2, exDates.size)
+            assertEquals(1420108210000L, exDates[0].time)
+            assertEquals(1435833020000L, exDates[1].time)
+        }
     }
 
     @Test
     fun testAndroidStringToRecurrenceSets_ZonedTimes() {
         // list of time zone times
-        val exDate = AndroidTimeUtils.androidStringToRecurrenceSet("${tzToronto.id};20150103T113030,20150704T113040",false) { ExDate(it) }
-        assertEquals(tzToronto, exDate.timeZone)
-        assertEquals(tzToronto.id, (exDate.getParameter(Parameter.TZID) as TzId).value)
-        val exDates = exDate.dates
-        assertEquals(Value.DATE_TIME, exDates.type)
-        assertEquals(tzToronto, exDates.timeZone)
+        AndroidTimeUtils.androidStringToRecurrenceSet(
+            "${tzToronto.id};20150103T113030,20150704T113040",
+            false
+        ) { ExDate(it) }.forEach { exDate ->
+            assertEquals(tzToronto, exDate.timeZone)
+            assertEquals(tzToronto.id, (exDate.getParameter(Parameter.TZID) as TzId).value)
+            val exDates = exDate.dates
+            assertEquals(Value.DATE_TIME, exDates.type)
+            assertEquals(tzToronto, exDates.timeZone)
+            assertEquals(2, exDates.size)
+            assertEquals(1420302630000L, exDates[0].time)
+            assertEquals(1436023840000L, exDates[1].time)
+        }
```

Comment:
When testing lists, `forEach` is not specific enough. The test may succeed, whether the method returns a list with one or with 100 items.